src/app/campaign/inventory/inventory.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
+import { InventoryZone } from '../../core';
+import { Flight, InventoryRollup } from '../store/models';
+
+@Component({
+  selector: 'grove-inventory',
+  template: `
+    <h2 class="title">Impressions</h2>
+    <p *ngIf="cantShowInventory(); else inventory">
+      Please select Start and End Dates, Series, and Zones to view inventory.
+    </p>
+    <ng-template #inventory>
+      <grove-goal-form [flight]="flight" (goalChange)="goalChange.emit($event)"></grove-goal-form>
+      <ul class="errors" *ngIf="errors as flightErrors">
+        <li class="error" *ngFor="let error of flightErrors"><mat-icon>priority_high</mat-icon> {{ error }}</li>
+      </ul>
+      <mat-divider></mat-divider>
+      <section>
+        <h3 class="title">{{ zoneNames }}</h3>
+        <grove-inventory-table *ngIf="rollup" [flight]="flight" [rollup]="rollup" [isPreview]="isPreview"></grove-inventory-table>
+      </section>
+    </ng-template>
+  `,
+  styleUrls: ['inventory.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InventoryComponent {
+  @Input() flight: Flight;
+  @Input() changed: boolean;
+  @Input() zones: InventoryZone[];
+  @Input() rollup: InventoryRollup;
+  @Input() isPreview: boolean;
+  @Input() previewError: any;
+  @Output() goalChange = new EventEmitter<Flight>();
+  zoneWeekExpanded = {};
+
+  get errors() {
+    return [
+      // allocation preview error
+      // TODO: Updated with discussed "nice_message" when available.
+      this.previewError && this.previewError.body && this.previewError.body.message,
```

Comment:
ok cool, it also displays any status message in the inventory.component. maybe we'll never have to do that error formatting/handling after all. agile!

Reply:
Even though it's going away, I found a bug in this preview error as well where I don't clear itself within the campaign because the state doesn't change on flight route change anymore, small fix incoming to track those errors in the store by flightId.